middleware/jwtAuthentication.js

```diff
@@ -3,7 +3,12 @@ const { verifyJwt } = require('../services/authentication')
 
 const authenticateJwt = async (req, res, next) => {
   const reqPath = req.path.toString()
-  if (reqPath !== config.auth.loginRoute && config.auth.isAuthEnabled) {
+  const reqMethod = req.method.toString()
+
+  const isProtectedUrl = config.auth.protectedUrls.find((request) => {
+    return (reqPath.startsWith(request.path) && (reqMethod === request.method))
```

Comment:
The start with is problematic. what if you have a path like /podcast which is unprotected and /padcast/save which is protected?

Reply:
I chose to check it by method and protected url for this reason...
Now if we have path like /podcast/save we would get post or put method and thats how it handles those cases.
The original problem was that we have multiple unprotected urls (/podcast/:any-number) and a few with specific methods that are protected, that is why I check what the current path requested is, and check if its starts with one of my protected urls with the relevant method.

Reply:
than I believe that in this case the protected url  in config should be like so :
/podcast/ -  the last / says anything that starts with /podcast 
and then you can check it using this module:
https://www.npmjs.com/package/path-to-regexp
this gives you the flexibility to decide whether its a regex or an absolute path